2-mandatory/2-financial-times.js

```diff
@@ -4,38 +4,79 @@
     The home page of the web site has a headline section, which only has space for article titles which are 65 characters or less.
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
+
 function potentialHeadlines(allArticleTitles) {
     // TODO
+    const headlines = [];
+    for (let articleTitle of allArticleTitles) {
+        if (articleTitle.length <= 65) {
+            headlines.push(articleTitle);
+        }
+    }
+    return headlines;
 }
 
+
+
+
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
     (you can assume words will always be seperated by a space)
 */
+
+
+
 function titleWithFewestWords(allArticleTitles) {
     // TODO
+    let shortestHeadlines = allArticleTitles[0];
+    // console.log(allArticleTitles[0])
+    for (let i = 1; i < allArticleTitles.length; i++) {
+        let currentTitle = allArticleTitles[i];
+        if (currentTitle.length < shortestHeadlines.length) {
+            shortestHeadlines = currentTitle  
+        }
+    }
+    return shortestHeadlines;
 }
 
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+// 
 function headlinesWithNumbers(allArticleTitles) {
     // TODO
+  let newArr = [];
+  for (let title of allArticleTitles) {
+    if (/[0-9]/.test(title) === true) {
+      newArr.push(title);
+    }
+  }
+
+  return newArr;
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
+
+
+
```

Comment:
This is a nice solution.
Can you think of another way to get the number of articles, without using the `numberOfArticles` variable?